### PR TITLE
Spring update

### LINKED ISSUE #105: Upgrade reactor to latest 3.2.x
We use the old 3.2.x reactor version: 3.2.5. We should upgrade to latest bugfix version (3.2.16 at the moment of writing).
Newer versions of reactor provide bugfixes.
Changelog: https://github.com/reactor/reactor-core/releases

### PR DESCRIPTION
**Pull request by deejay1**
Update Spring Boot to 2.3.4 and related dependencies.
Fixes #105 by upgrading reactor to 3.3.10

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -124,9 +114,11 @@ subprojects {
     }
 
     dependencies {
+        implementation platform(group: 'org.springframework.boot', name: 'spring-boot-dependencies', version: versions.spring_boot)
```
Comment (pzmi) [inline build.gradle:117]: I'm not so sure about using spring-boot-dependencies here. envoy-control-runner is just a small and non-crucial part of envoy-control. By using spring-boot dependencies in other subprojects we make them dependent on Spring Boot.

Comment (deejay1) [reply build.gradle:117]: Only on its bill of materials, of course I can add them manually if that will be the chosen way.

**Review by pzmi — APPROVED** [review]

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoycontrol/EnvoyControlTestApp.kt [hunk 2]
```diff
@@ -228,7 +228,7 @@ class EnvoyControlRunnerTestApp(
 
 data class Health(
     val status: Status,
-    val details: Map<String, HealthDetails>
+    val components: Map<String, HealthDetails>
```
Comment (slonka) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoycontrol/EnvoyControlTestApp.kt:231]: This rename seems like an unrelated change. Can you explain?

**Review by slonka — APPROVED** [review]
Looks alright, just one question.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoycontrol/EnvoyControlTestApp.kt (hunk 2 above)
Comment (deejay1) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/envoycontrol/EnvoyControlTestApp.kt:231]: Spring changed the format of the health endpoint somewhere between 2.1 and 2.3 and details is gone now.
